pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/KitCheckService.java

```diff
@@ -111,39 +111,42 @@ public KitCheckResult checkForInitialKits() {
         return kitCheckResult;
     }
 
-    private void processPotentialKitRecipient(String studyGuid, long kitTypeId,
+    public KitCheckResult processPotentialKitRecipient(String studyGuid, long kitTypeId,
                                               KitCheckResult kitCheckResult,
                                               KitConfiguration kitConfiguration,
-                                              PotentialRecipient candidate) {
+                                              PotentialRecipient candidate, boolean event) {
         String userGuid = candidate.getUserGuid();
 
         if (candidate.getAddressId() == null) {
             log.warn("Participant {} is missing a default mailing address", userGuid);
-            return;
+            return null;
         }
 
         if (candidate.getAddressValidationStatus() == null
                 || candidate.getAddressValidationStatus() == DSM_INVALID_ADDRESS_STATUS) {
             log.warn("Participant {} has an invalid mailing address", userGuid);
-            return;
+            return null;
```

Comment:
Does it make sense to return more information about the specific "failure" cases here in the `KitCheckResult` instead of relying on the logging alone and returning `null`?

Reply:
From event, we already look for address in CreateKitEventAction and log error.
Here in KitCheckService it can be warn because CheckKitsJob will retry these later.
Moreover even if we return some error info , Housekeeping can only log I guess.